src/lib/features/client-feature-toggles/client-feature-toggle.controller.ts

```diff
@@ -191,22 +192,34 @@ export default class FeatureController extends Controller {
                 const sortedToggles = features.sort((a, b) =>
                     a.name.localeCompare(b.name),
                 );
-                const sortedNewToggles = delta?.updated.sort((a, b) =>
-                    a.name.localeCompare(b.name),
-                );
+                if (delta?.events[0].type === 'hydration') {
+                    const hydrationEvent: DeltaHydrationEvent =
+                        delta?.events[0];
+                    const sortedNewToggles = hydrationEvent.features.sort(
+                        (a, b) => a.name.localeCompare(b.name),
+                    );
 
-                if (
-                    !this.deepEqualIgnoreOrder(sortedToggles, sortedNewToggles)
-                ) {
-                    this.logger.warn(
-                        `old features and new features are different. Old count ${
-                            features.length
-                        }, new count ${delta?.updated.length}, query ${JSON.stringify(query)},
+                    if (
+                        !this.deepEqualIgnoreOrder(
+                            sortedToggles,
+                            sortedNewToggles,
+                        )
+                    ) {
+                        this.logger.warn(
+                            `old features and new features are different. Old count ${
+                                features.length
+                            }, new count ${hydrationEvent.features.length}, query ${JSON.stringify(query)},
                         diff ${JSON.stringify(
                             diff(sortedToggles, sortedNewToggles),
                         )}`,
+                        );
+                    }
+                } else {
+                    this.logger.warn(
+                        `Delta diff should have only hydration event, query ${JSON.stringify(query)}`,
```

Comment:
Is this the correct implementation of the diff checker? If we only compare on hydration event, how do we know that the deltas are producing the correct result?

Reply:
Well, every time you add new events to cache, they will instantly applied to hydration event. 
We only take features from database at first, but after that we will start manipulation the hydration event every second.